opm/simulators/linalg/PressureSolverPolicy.hpp

```diff
@@ -60,12 +60,16 @@ namespace Amg
             PressureInverseOperator(Operator& op, const boost::property_tree::ptree& prm, const Comm& comm)
                 : linsolver_()
             {
-                if (op.category() == Dune::SolverCategory::overlapping) {
-                    linsolver_.reset(new Solver(op.getmat(), comm, prm, std::function<X()>()));
-                } else {
-                    linsolver_.reset(new Solver(op.getmat(), prm, std::function<X()>()));
-                }
+                assert(op.category() == Dune::SolverCategory::overlapping);
```

Comment:
This is of course an improvement compared to previous version, but I am wondering whether an assert is the right tool.Maybe we should change the template parameter instead?

```
template<class GI, class LI>
PressureInverseOperator(Operator& op, const boost::property_tree::ptree& prm, const Dune::OwnerOverlapCommunication<GI,LI>& comm);
````
Seems like that was what we actually wanted to achieve (test whether this is parallel). Original version did not do that of course.
If the check for overlapping was for another person, we should in addition throw an excption if it does not hold
￼                : linsolver_()

Reply:
I think your idea is good. Also, @akva2 suggested to replace the shared_ptrs with unique_ptrs.

If you think the current PR is ok enough, could we make this a second PR? (Testing it would be quicker of course with this PR in place.)

Reply:
ok